### PR TITLE
User Invite Functionality 

### LINKED ISSUE #122: [FE] User Invite
### Description
Admin want to be able to invite users via the User Management page.

Implement the Figma design to create/invite a new user and connect it to the backend user API. As well, create a new "Invite pending/accepted" visual in the Status column.

To invite a user, we must first create a new user and assign them an `email`,  `role`, and set `acceptedInvite: false`. To create a new user, send a `POST` to `/api/users` with the above information in the request body.

Once a user is successfully invited (created), be sure to update the user management page (through browser reload or state update) so the newly invited user is displayed in the table. Display a toast to show successful invitation or error.

- Invite pending text is displayed if `acceptedInvite: false`
- Toggle off if `isActive: false` 
- Toggle on if `isActive: true`
- use `react-tooltip` (already installed) to display appropriate text when hovering
    - the styling for the tooltip can be copied from `RecentLogs.jsx`

**Important notes** ->

- ensure that the email users are inviting does not already exist in the database
- As shown in the design, accounts should be sorted in the following order:
    - Active, Pending invite, deactivated
    - This sort can be done client-side

### Acceptance Criteria
- [ ]  Proper error handling
- [ ]  Proper JS doc comments above functions
- [ ]  Test cases are met

### Other Notes
- N/a

### PR DESCRIPTION
**Pull request by ShmulTomer**
## User Invite Functionality

Issue Number(s): 122

What does this PR change and why?
- invite user button + modal
- error handling
- new backend POST to add a user (adds a user with name “Invited User” so the table column has an actual value)
- Status depends on value of acceptedInvite and isActive
- Also previously, changing the access level through the table refreshed the page, but now it does a table state update w/o refresh

### Checklist

- [x] Requirements have been implemented
- [ ] Acceptance criteria is met
- [ ] Database schema [docs](https://www.notion.so/gtbitsofgood/Database-Schema-Docs-fd377ded10b54890b65fd99025488cee?pvs=4) have been updated or are not necessary
- [ ] Relevant reviewers (EM) have been assigned to this PR

### Critical Changes
for the user API schema (used for createUser), I changed
emailVerified: {
    type: Boolean,
    default: null,   >>> default: false,
  },

### How To Test

change values through MongoDB Compass to see the full changes (since the seed values have all “acceptedInvite” false); in the Users table, make some of the users “acceptedInvite” to true

### CODE REVIEW

**Comment by ShmulTomer** [comment]
Some thoughts / questions:
- maybe there could be a better indicator that theres more pages on the User Management page? Took me a sec to figure out where my new entry went
- also should the table rows be listed in reverse order? With the newest users showing up on top?

File: src/components/User/UserTable.jsx [hunk 1]
```diff
@@ -1,18 +1,21 @@
...
   AtSymbolIcon,
   ClipboardIcon,
+  PlusIcon,
 } from "@heroicons/react/24/solid";
 import LoadingAnimation from "../LoadingAnimation";
 import toast from "react-hot-toast";
 import { consts } from "@/utils/consts";
 import ToggleSwitch from "./ToggleSwitch"
 import ConfirmCancelModal from "../ConfirmCancelModal";
+import { set } from "mongoose";
```
Comment (afazio1) [inline src/components/User/UserTable.jsx]: remove if not needed

File: src/pages/api/users/index.js [hunk 2]
```diff
@@ -16,6 +18,53 @@ export default async function handler(req, res) {
         message: e.message,
       });
     }
+  } else if (req.method == "POST") {
```
Comment (afazio1) [inline src/pages/api/users/index.js:23]: Only "Admin" and "Manager" roles can invite users, so make sure to check the session here.
You can use NextAuth's `getServerSideSession()`, which will give you `session.user.role`

**Review by afazio1 — CHANGES_REQUESTED** [review]
Looks good 👍 See comments for a minor change and adding authorization on the backend

**Review by laurenmcp — CHANGES_REQUESTED** [review]
Nice job on the functionality! Small nit for design, it would be great to decrease the width of the invite a user button to match the figma. 
Also, I'm assuming a tag is used to display the selected user role (rather than the figma design) due to technical limitations. We need to double check with Hemadri to see if this is OK from a UI perspective. We may need to implement the dropdown to change `DropDownMenuOption` and `DropDownMenu` labels conditionally, rather than use the tag approach.

File: src/components/User/UserTable.jsx (hunk 1 above)
Comment (ShmulTomer) [reply src/components/User/UserTable.jsx]: Removed

File: src/pages/api/users/index.js (hunk 2 above)
Comment (ShmulTomer) [reply src/pages/api/users/index.js:23]: Done; I think this also needs to be implemented for other backend functions as well

**Comment by ShmulTomer** [comment]
Ok, I addressed all the changes @laurenmcp @afazio1 

To get the Dropdown to look like the Figma, Im dynamically changing the label from the Modal class, since the DropdownMenu doesnt have the ability to change labels to the selected option by itself

**Review by afazio1 — APPROVED** [review]

**Review by afazio1 — APPROVED** [review]

**Pull request merged**